static/src/javascripts/projects/common/modules/commercial/third-party-tags/outbrain.js

```diff
@@ -16,109 +18,152 @@ define([
+        });
+    }
+
+    function tracking(widgetCode) {
+        // Ophan
+        require(['ophan/ng'], function (ophan) {
+            ophan.record({
+                outbrain: {
+                    widgetId: widgetCode
                 }
+            });
+        });
+    }
 
-                this.tracking(widgetCode);
-                require(['js!' + outbrainUrl]);
-            }.bind(this));
-        },
+    function trackAd(id) {
+        return new Promise(function (resolve, reject) {
+            var onAdLoaded = function (event) {
+                if (event.slot.getSlotElementId() === id) {
+                    off();
+                    resolve(!event.isEmpty);
+                }
+            };
 
-        tracking: function (widgetCode) {
-            // Ophan
-            require(['ophan/ng'], function (ophan) {
-                ophan.record({
-                    outbrain: {
-                        widgetId: widgetCode
-                    }
-                });
-            });
-        },
+            var onAllAdsLoaded = function () {
+                off();
+                reject(new Error('Unable to load Outbrain widget: slot ' + id + ' was never loaded'));
+            };
 
-        getSection: function () {
-            return config.page.section.toLowerCase().match('news')
-                || contains(['politics', 'world', 'business', 'commentisfree'], config.page.section.toLowerCase()) ? 'sections' : 'all';
-        },
+            function off() {
+                mediator.off('modules:commercial:dfp:rendered', onAdLoaded);
+                mediator.off('modules:commercial:dfp:alladsrendered', onAllAdsLoaded);
+            }
 
-        identityPolicy: function () {
-            return (!identity.isUserLoggedIn() || !(identity.isUserLoggedIn() && config.page.commentable));
-        },
+            mediator.on('modules:commercial:dfp:rendered', onAdLoaded);
+            mediator.on('modules:commercial:dfp:alladsrendered', onAllAdsLoaded);
+        });
+    }
 
-        hasHighRelevanceComponent: function () {
-            return detect.adblockInUse() || config.page.edition.toLowerCase() === 'int';
-        },
+    function identityPolicy() {
+        return !(identity.isUserLoggedIn() && config.page.commentable);
+    }
+
+    /*
+        Loading Outbrain is dependent on successful return of high relevance component
+        from DFP. AdBlock is blocking DFP calls so we are not getting any response and thus
+        not loading Outbrain. As Outbrain is being partially loaded behind the adblock we can
+        make the call instantly when we detect adBlock in use.
+     */
+    function loadInstantly() {
+        return !document.getElementById('dfp-ad--merchandising-high') ||
+            detect.adblockInUse();
+    }
+
+    function init() {
+        if (config.switches.outbrain
+            && !config.page.isFront
+            && !config.page.isPreview
+            && config.page.section !== 'childrens-books-site'
```

Comment:
From our discussion with @jbreckmckye a few months ago:

> Policies would normally describe particular states of the page, rather than individual features. For example, we would have a userLoggedIn policy or a sensitivePage policy (is Outbrain supposed to be disabled on all sensitive content, incidentally?). This allows us to describe what the policies in these cases are. Otherwise I don't think we really take advantage of centralizing this logic.

Just something to think about before you will do that.